h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1209,6 +1217,95 @@ else if (orows instanceof long[]) {
     }
   }
 
+  /**
+   * @return frame without rows with NAs in `columnIndex` column
+   */
+  public Frame filterOutNAsInColumn(int columnIndex) {
+    Frame noNaPredicateFrame = new IsNotNaTask().doAll(1, Vec.T_NUM, new Frame(this.vec(columnIndex))).outputFrame();
+    return selectByPredicate(noNaPredicateFrame);
+  }
+
+  /**
+   * @return frame with all the rows except for those whose value in the `columnIndex' column equals to `value`
+   */
+  public Frame filterNotByValue(int columnIndex, double value) {
+    return filterByValueBase(columnIndex, value, true);
+  }
+
+  /**
+   * @return frame with all the rows whose value in the `columnIndex' column equals to `value`
+   */
+  public Frame filterByValue(int columnIndex, double value) {
+    return filterByValueBase(columnIndex, value,false);
+  }
+
+  private Frame filterByValueBase(int columnIndex, double value, boolean isInverted) {
+    Frame predicateFrame = new FilterByValueTask(value, isInverted).doAll(1, Vec.T_NUM, new Frame(this.vec(columnIndex))).outputFrame();
+    return selectByPredicate(predicateFrame);
+  }
+
+  private Frame selectByPredicate(Frame predicateFrame) {
+    String[] names = names().clone();
+    byte[] types = types().clone();
+    String[][] domains = domains().clone();
+
+    add("predicate", predicateFrame.anyVec());
+    Frame filtered = new Frame.DeepSelect().doAll(types, this).outputFrame(Key.<Frame>make(), names, domains); // TODO here we do a copy of the Frame because of the outputFrame. probably can do better.
+    predicateFrame.delete();
+    remove("predicate");
+    return filtered;
+  }
+
+  /**
+   *
+   * @param columnIndex
+   * @return frame with specified column encoded into binary enumeration column
+   */
+  public Frame vectorAsQuasiBinomial(int columnIndex) {
+
+    Vec columnToEncode = vec(columnIndex);
+    String[] domains = columnToEncode.domain().clone();
+    Frame encodedFrame = new AsQuasiBinomialTask(domains).doAll( Vec.T_NUM, columnToEncode).outputFrame(); // TODO how to avoid creation of Frame when doing MRT?
+    replace(columnIndex, encodedFrame.vec(0));
+    DKV.put(this);
+    columnToEncode.remove();
+    return this;
+  }
+
+  /** return a frame with unique values from the specified column */
+  public Frame uniqueValuesBy(int columnIndex) {
```

Comment:
@michalkurka review later

Reply:
@michalkurka just a reminder. We were discussing that there are at least two common approaches to using folds :

1. specify name of the fold column and make sure that datasets already have this column 
2. specify only number of folds and let us generate folds for user ( in this case we don't need this method for calculating number of folds)

Current implementation support only first method for now